File: isabelle.eclipse.ui/src/isabelle/eclipse/ui/IsabelleImages.java
package isabelle.eclipse.ui;

import java.net.MalformedURLException;
import java.net.URL;

import org.eclipse.jface.resource.ImageDescriptor;
import org.eclipse.jface.resource.ImageRegistry;
import org.eclipse.swt.graphics.Image;


public class IsabelleImages {

	/* Declare Common paths */
	private static URL ICON_BASE_URL= null;

	static {
		String pathSuffix = "icons/"; //$NON-NLS-1$	
		ICON_BASE_URL= IsabelleUIPlugin.getDefault().getBundle().getEntry(pathSuffix);
	}
	
	public static final String IMG_RAW_OUTPUT_CONSOLE = ICON_BASE_URL + "isabelle.png";
	public static final String IMG_CONTENT_ASSIST = ICON_BASE_URL + "isabelle.png";
	public static final String IMG_OUTLINE_ITEM = ICON_BASE_URL + "isabelle.png";
	
	/**
	 * Returns the <code>Image<code> identified by the given path,
	 * or <code>null</code> if it does not exist.
	 */
	public static Image getImage(String path) {
		ImageRegistry imageRegistry = IsabelleUIPlugin.getDefault().getImageRegistry();
		Image image = imageRegistry.get(path);
		if (image == null) {
			getImageDescriptor(path);
			image = imageRegistry.get(path);
		}
		
		return image;
	}
	
	/**
	 * Returns the <code>ImageDescriptor<code> identified by the given path,
	 * or <code>null</code> if it does not exist.
	 */
	public static ImageDescriptor getImageDescriptor(String path) {
		ImageRegistry imageRegistry = IsabelleUIPlugin.getDefault().getImageRegistry();
		ImageDescriptor desc = imageRegistry.getDescriptor(path);
		if (desc == null) {
			desc = ImageDescriptor.getMissingImageDescriptor();
			try {
				desc = ImageDescriptor.createFromURL(new URL(path));
				imageRegistry.put(path, desc);
			} catch (MalformedURLException me) {
			}
		}
		
		return desc;
	}
	
}

File: isabelle.eclipse.ui/src/isabelle/eclipse/ui/IsabelleImages.scala
package isabelle.eclipse.ui

import java.net.{MalformedURLException, URL}

import org.eclipse.jface.resource.ImageDescriptor


/**
 * Isabelle image definitions.
 *
 * When images are used in label providers (e.g. where Image) is required, they must be disposed manually.
 * For convenience, [[org.eclipse.jface.resource.ResourceManager]] could be used.
 *
 * @author Andrius Velykis
 */
object IsabelleImages {

  private lazy val ICON_BASE_URL = IsabelleUIPlugin.getDefault.getBundle.getEntry("icons/")

  val MISSING_ICON = ImageDescriptor.getMissingImageDescriptor

  lazy val ISABELLE_FILE = create("isabelle_file.gif")
  lazy val ISABELLE_LOADED_FILE = create("isabelle_loaded_file.gif")
  
  lazy val RAW_OUTPUT_CONSOLE = create("isabelle.png")
  lazy val CONTENT_ASSIST = create("isabelle.png")
  lazy val OUTLINE_ITEM = create("isabelle.png")
  lazy val RAW_TREE = create("raw_tree.gif")
  lazy val SHOW_TRACE = create("show_trace.gif")


  private def create(iconPath: String) = {
    try {
      val url = new URL(ICON_BASE_URL, iconPath)
      ImageDescriptor.createFromURL(url)
    } catch {
      case _: MalformedURLException => MISSING_ICON
    }
  }

}
